app/assets/javascripts/sitewide/file_preview.js

```diff
@@ -349,8 +353,12 @@
             }
           }
         } else {
+          modal.find('.file-edit-link').css('display', 'none');
           modal.find('.file-preview-container').html(data['preview-icon']);
         }
+        if (readOnly) {
```

Comment:
I think it would be better to send `readOnly` as a parameter to `file-preview-url` and to have this `edit permission` logic all in one place.

Reply:
This parameter is used to disable file editing on the whole page(like protocol preview), and not per file.